lib/rules/function-calc-no-unspaced-operator/index.mjs

```diff
@@ -70,169 +74,306 @@ const rule = (primary, _secondaryOptions, context) => {
-			 * @param {number} index
-			 * @param {'before' | 'after'} position
-			 * @returns {void}
-			 */
-			function checkWhitespaceAroundOperator(operatorNode, parent, index, position) {
-				const isBefore = position === 'before';
-				const aroundNode = parent.value[index + (isBefore ? -1 : 1)];
+			const tokens = tokenize({ css: value });
 
-				if (isWhitespaceNode(aroundNode)) {
-					const whitespace = getWhitespace(aroundNode);
+			{
+				// Step 1
+				// Step 1.1
+				// Re-tokenize dimensions with units containing dashes.
+				// These might be typo's.
+				// For example: `10px-20px` has a unit of `px-20px`
+				for (let i = 0; i < tokens.length; i++) {
+					const token = tokens[i];
 
-					if (whitespace === ' ') return;
+					// NOTE:
+					// This is a mess, I am fairly certain that it is correct, but too much is happening with strings.
+					// This will be brittle and might break when the css-tokenizer is updated.
 
-					const indexOfFirstNewLine = whitespace.search(NEWLINE_REGEX);
+					// Note:
+					// This will produce invalid CSS for exotic units (e.g. `1\\23`).
+					// No such units are specified today.
+					// Correctly serializing units would resolve any issues.
 
-					if (indexOfFirstNewLine === 0) return;
+					if (!token || token[0] !== TokenType.Dimension) continue;
 
-					if (context.fix) {
-						needsFix = true;
+					if (token[4].unit.startsWith('--')) continue;
 
-						const [token] = aroundNode.value;
+					const indexOfDash = token[4].unit.indexOf('-');
 
-						if (token) {
-							token[1] = indexOfFirstNewLine === -1 ? ' ' : whitespace.slice(indexOfFirstNewLine);
-						}
+					if (indexOfDash === -1) continue;
 
-						return;
-					}
-				} else if (
-					isTokenNode(aroundNode) &&
-					(isOperandNode(aroundNode) || isScssInterpolation(aroundNode, parent))
-				) {
-					if (context.fix) {
-						needsFix = true;
+					const remainder = token[4].unit.slice(indexOfDash);
 
-						if (isBefore) {
-							aroundNode.value[1] += ' ';
-						} else {
-							aroundNode.value[1] = ` ${aroundNode.value[1]}`;
-						}
+					if (remainder.length === 1) continue;
 
-						return;
-					}
-				} else {
-					return;
-				}
+					token[4].unit = token[4].unit.slice(0, indexOfDash);
+					token[1] = `${token[4].signCharacter === '+' ? '+' : ''}${token[4].value}${
+						token[4].unit
+					}`;
+					token[3] = token[2] + token[1].length;
 
-				const delimToken = getDelimToken(operatorNode);
+					const remainderTokens = tokenize({ css: remainder }).slice(0, -1); // Trim EOF token
 
-				if (!delimToken) return;
+					remainderTokens.forEach((remainderToken) => {
+						remainderToken[2] += token[3];
+						remainderToken[3] += token[3];
+					});
 
-				const { value: operator, startIndex } = delimToken;
-				const message = messages[isBefore ? 'expectedBefore' : 'expectedAfter'](operator);
+					tokens.splice(i + 1, 0, ...remainderTokens);
```

Comment:
Could you check that we didn't regress #6045?
see https://github.com/stylelint/stylelint/issues/6044#issuecomment-1112878879
It seems to be covered by https://github.com/stylelint/stylelint/pull/7655/files#diff-4aac0dd17733f1428b79c71d6c9d05c314d49cf340415478c12bf5d9dfd330c6R225 but I am being cautious.